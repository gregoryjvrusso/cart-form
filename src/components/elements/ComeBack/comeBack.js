import React from "react";
import Flex from "../Flex";
import { ArrowLeft, TextComeBack } from "./comeBack.styles";
import { arrowWhite } from "../../../temp/index";
import Box from "../Box";

const ComeBack = () => (
  <Flex
    width="100%"
    justifyContent={["flex-start", "flex-start", "center"]}
    alignItems="center"
  >
    <ArrowLeft alt="botÃ£o voltar" src={arrowWhite} />
    <Flex
      width="100%"
      justifyContent="center"
      alignItems="center"
      display={["flex", "flex", "none"]}
    >
      <TextComeBack>
        <span>Etapa 2</span> de 3
      </TextComeBack>
    </Flex>
    <Box display={["none", "none", "block"]}>
      <TextComeBack>Alterar forma de pagamento</TextComeBack>
    </Box>
  </Flex>
);

export default ComeBack;
