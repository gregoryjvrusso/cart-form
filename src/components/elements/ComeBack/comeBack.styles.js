import styled from "styled-components";
import { theme } from "styled-tools";
import Text from "../Text";

export const ArrowLeft = styled.img`
  transform: rotate(90deg);
  margin-right: ${theme("spaces.2")}px;
`;

export const TextComeBack = styled(Text)`
  color: ${theme("colors.texts.white")};
  font-size: ${theme("fontSizes.1")}px;
  text-align: center;
  span {
    font-weight: ${theme("fontWeights.bold")};
  }
`;
