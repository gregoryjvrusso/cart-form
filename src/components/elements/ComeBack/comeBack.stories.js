import React from "react";
import { storiesOf } from "@storybook/react";
import { ThemeProvider } from "styled-components";
import Flex from "../Flex";
import theme from "../../../common/theme";

import ComeBack from "../ComeBack";

storiesOf("Elements/ComeBack", module).add("default", () => (
  <ThemeProvider theme={theme}>
    <Flex bg="red">
      <ComeBack />
    </Flex>
  </ThemeProvider>
));
