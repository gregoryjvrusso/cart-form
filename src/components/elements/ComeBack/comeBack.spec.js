import React from "react";
import { render } from "@testing-library/react";
import ComeBack from "../ComeBack";

describe("ComeBack component", () => {
  it("deve conseguir renderizar [snapshot]", async () => {
    const { container } = render(
      <ComeBack />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
