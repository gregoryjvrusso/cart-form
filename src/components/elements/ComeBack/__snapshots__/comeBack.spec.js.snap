// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComeBack component deve conseguir renderizar [snapshot] 1`] = `
<div
  class="sc-AxjAm hVkxHI sc-AxirZ gLMeKs"
  data-testid="flex"
  width="100%"
>
  <img
    alt="botÃ£o voltar"
    class="sc-AxhCb cmyoQP"
    src="arrow-white.svg"
  />
  <div
    class="sc-AxjAm fSzghi sc-AxirZ hzUFUl"
    data-testid="flex"
    display="flex,flex,none"
    width="100%"
  >
    <p
      class="sc-AxiKw dneoXV sc-AxhUy jEDGwC"
      data-testid="text"
    >
      <span>
        Etapa 2
      </span>
       de 3
    </p>
  </div>
  <div
    class="sc-AxjAm kvAEpy"
    data-testid="box"
    display="none,none,block"
  >
    <p
      class="sc-AxiKw dneoXV sc-AxhUy jEDGwC"
      data-testid="text"
    >
      Alterar forma de pagamento
    </p>
  </div>
</div>
`;
